File: frontend/src/api/targetProfileDraft.ts
export interface TargetProfileDraft {
  source: 'OPENAPI' | 'README'
  suggestedTargetId: string
  suggestedTargetName: string
  suggestedBaseUrl: string | null
  readOnlyOperations: DraftReadOnlyOperation[]
  yaml: string
  warnings: string[]
}

export interface DraftReadOnlyOperation {
  id: string
  title: string
  description: string
  path: string
  expectedStatusCodes: number[]
}

File: src/main/kotlin/com/project/agenticreliabilitylab/targetprofiledraft/domain/TargetProfileDraft.kt
package com.project.agenticreliabilitylab.targetprofiledraft.domain

enum class TargetProfileDraftSource {
    OPENAPI,
    README,
}

data class TargetProfileDraft(
    val source: TargetProfileDraftSource,
    val suggestedTargetId: String,
    val suggestedTargetName: String,
    val suggestedBaseUrl: String?,
    val readOnlyOperations: List<DraftReadOnlyOperation>,
    val yaml: String,
    val warnings: List<String>,
)

data class DraftReadOnlyOperation(
    val id: String,
    val title: String,
    val description: String,
    val path: String,
    val expectedStatusCodes: Set<Int>,
)

class TargetProfileDraftException(message: String, cause: Throwable? = null) :
    IllegalArgumentException(message, cause)
